prediction_market_agent/agents/arbitrage_agent/deploy.py

```diff
@@ -0,0 +1,193 @@
+import typing as t
+
+from langchain_core.output_parsers import PydanticOutputParser
+from langchain_core.prompts import PromptTemplate
+from langchain_core.runnables import RunnableSerializable
+from langchain_openai import ChatOpenAI
+from prediction_market_agent_tooling.deploy.agent import (
+    MAX_AVAILABLE_MARKETS,
+    DeployableTraderAgent,
+)
+from prediction_market_agent_tooling.gtypes import Probability
+from prediction_market_agent_tooling.markets.agent_market import (
+    AgentMarket,
+    FilterBy,
+    SortBy,
+)
+from prediction_market_agent_tooling.markets.data_models import (
+    BetAmount,
+    Position,
+    ProbabilisticAnswer,
+    TokenAmount,
+    Trade,
+    TradeType,
+)
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.markets.omen.omen import OmenAgentMarket
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.langfuse_ import (
+    get_langfuse_langchain_config,
+    observe,
+)
+
+from prediction_market_agent.agents.arbitrage_agent.data_models import (
+    CorrelatedMarketPair,
+    Correlation,
+)
+from prediction_market_agent.db.pinecone_handler import PineconeHandler
+from prediction_market_agent.utils import APIKeys
+
+
+class DeployableOmenArbitrageAgent(DeployableTraderAgent):
+    """Agent that places mirror bets on Omen for (quasi) risk-neutral profit."""
+
+    correlation_threshold: float = 0.8
+
+    def load(self) -> None:
+        self.subgraph_handler = OmenSubgraphHandler()
```

Comment:
Any reason you've made this agent omen-specific? Looks like the only thing you've used the sgh for is to get markets. So it looks like this could easily be changed to become market-platform agnostic

Reply:
I made it Omen-specific because the `related_markets` should be queryable via Pinecone (because of the TTA agent), and only Omen agents are stored there currently.

Reply:
Indexing occurs below
https://github.com/gnosis/prediction-market-agent/blob/a1931b8be64cbc858a139059279b971485064358/prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py#L190

I don't think it prevents someone from running the agent - you just need Pinecone api keys and, on the first run, it will fill out Pinecone and have the same dataset as we do.

> Regardless, there doesn't seem to be a need to have any Omen dependencies here - you could just have a if market_type != MarketType.Omen: raise ... check

Makes sense to me, implementing this.

Reply:
Well it does if the user wants to run just this agent, and not the DeployableThinkThoroughlyAgent as well! I think we need to do `self.pinecone_handler.insert_all_omen_markets_if_not_exists(...)` somewhere here

Reply:
Just had a look at what `pinecone_handler.update_markets` is doing. Is this what we want? Two things I'm not sure about:
1. It doesn't look like it's removing closed markets from the index
2. It only fetches the last 7 days' worth of created markets `created_after = utcnow() - datetime.timedelta(days=7)`. Don't we want all open markets?

Reply:
Implemented changes as suggested by @evangriffiths :
-> Removing closing markets (see pinecone_handler::remove_resolved_markets)
-> Fetching all open markets instead of markets created in the last 7 days (pinecone_handler::insert_all_omen_markets_if_not_exists)